config/config_encryption.go

```diff
@@ -188,17 +189,16 @@ func getScryptDK(key, salt []byte) ([]byte, error) {
 	return scrypt.Key(key, salt, 32768, 8, 1, 32)
 }
 
-func makeNewSessionDK(key []byte) ([]byte, error) {
-	var err error
+func makeNewSessionDK(key []byte) (dk, storedSalt []byte, err error) {
 	storedSalt, err = crypto.GetRandomSalt([]byte(SaltPrefix), SaltRandomLength)
 	if err != nil {
-		return nil, err
+		return nil, storedSalt, err
 	}
 
-	dk, err := getScryptDK(key, storedSalt)
+	dk, err = getScryptDK(key, storedSalt)
 	if err != nil {
-		return nil, err
+		return nil, storedSalt, err
```

Comment:
This is really minor, but is there any reason to return the salt here? If there's an error on the other side, this storedSalt does not get used

Reply:
Done.